### ISSUE TITLE
Grid display for system tray icons

### ISSUE THREAD

**Issue by Plueres** [ramensoftware/windows-11-taskbar-styling-guide#40]
Make the system tray icons align in a grid where you can change the column or row amount, if there are a lot of tray icons. 

This was possible in windows 10 but I'm not sure if it is possible in windows 11.

**Comment by bbmaster123** [comment]
sorry for the delay
I've looked into this, but it doesn't look like this can be changed in the taskbar styler mod, at least not without manually adjusting each icon manually, which would probably cause some issues.

We would need to be able to alter the behavior of the stackpanel to tell it how many rows/columns to use, but I don't see a way of doing that, even with uwpspy. we would likely need a new mod to do that on the win11 taskbar. 

still possible with the win10 taskbar though if you use EP or SAB

**Comment by jpedrelli** [comment]
I would be happy to have 2 lines of icons, but a configurable grid would be great

![Windows 11 Icon tray](https://github.com/user-attachments/assets/6e8cdd2b-62f2-4804-b74f-1ce73ddd324c)

![Windows 11 Icon tray 2 lines](https://github.com/user-attachments/assets/61ad1f9f-0ac7-4432-9b64-bea2a2378a15)

**Comment by Plueres** [comment]
> sorry for the delay
> I've looked into this, but it doesn't look like this can be changed in the taskbar styler mod, at least not without manually adjusting each icon manually, which would probably cause some issues.
> 
> We would need to be able to alter the behavior of the stackpanel to tell it how many rows/columns to use, but I don't see a way of doing that, even with uwpspy. we would likely need a new mod to do that on the win11 taskbar. 
> 
> still possible with the win10 taskbar though if you use EP or SAB

Would this be possible with explorerpatcher with the w10 taskbar? I currently use it, and I can not change the grid size separately. Also, what is SAB?

**Comment by bbmaster123** [comment]
yes it should be possible with EP using the w10 taskbar, with a registry tweak [https://winaero.com/how-to-show-notification-area-icons-in-two-rows-in-windows-11/](https://winaero.com/how-to-show-notification-area-icons-in-two-rows-in-windows-11/)

SAB is startallback, see here [https://www.startallback.com](https://www.startallback.com)

Its separate from explorerpatcher, has a 30 day trial period, and might conflict if you try to use both at the same time, depending on settings
If I recall, it has an option for 2 rows built right in.

**Comment by Plueres** [comment]
Thanks for the link, I have been searching on where I found it before, but it stopped working some time around 2022, I think. This is unfortunately not working for me anymore. 

I think I'll give startallback a go, but I'm kind of worried it will also intervene with my installation of start11. I'll check that one out and see if it has it. 

But that link of winaero was kind of the reason I asked this question myself, because I couldn't find it anywhere

**Comment by bbmaster123** [comment]
thats a shame, I figured since its the windows 10 taskbar the registry tweak would still work

**Comment by Plueres** [comment]
Unfortunately not. Though I did try startallback and even though it doesn't have an option to change the grid of the tray icons (in the stable version at least), I'll still switch over to it because it has better customization features imo

**Comment by m417z** [comment]
[Taskbar tray icon spacing and grid](https://windhawk.net/mods/taskbar-notification-icon-spacing) v1.2 now has the option to have a grid of tray icons.

<img width="399" height="48" alt="Image" src="https://github.com/user-attachments/assets/a6baed47-b888-4c68-a697-b7b9c8fecddd" />
